src/Symfony/Component/Security/Core/Authorization/AuthorizationChecker.php

```diff
@@ -34,6 +34,13 @@ class AuthorizationChecker implements AuthorizationCheckerInterface
 
     public function __construct(TokenStorageInterface $tokenStorage, AuthenticationManagerInterface $authenticationManager, AccessDecisionManagerInterface $accessDecisionManager, bool $alwaysAuthenticate = false, bool $exceptionOnNoToken = true)
     {
+        if (false !== $alwaysAuthenticate) {
+            trigger_deprecation('symfony/security-core', '5.4', 'Not setting the 4th argument of "%s" to "false" is deprecated.', __METHOD__);
```

Comment:
(minor) we might prefer talking about arguments names over positions in deprecation messages, see e.g. https://github.com/symfony/symfony/commit/8e3058d95acc7d177516bfacce794f64695c2b0d